pkg/storage/s3/s3.go

```diff
@@ -47,26 +57,82 @@ func NewDriver(c *imageregistryv1.ImageRegistryConfigStorageS3, kubeconfig *rest
 	}
 }
 
+// GetConfig reads configuration for the S3 cloud platform services.
+func GetConfig(kubeconfig *rest.Config, listers *regopclient.Listers) (*S3, error) {
+	cfg := &S3{}
+
+	infra, err := util.GetInfrastructure(listers)
+	if err != nil {
+		return nil, err
+	}
+
+	if infra.Status.PlatformStatus.Type == configapiv1.AWSPlatformType {
```

Comment:
We should already know by this point if we are using AWS/S3, should this just be
`if infra.Status.PlatformStatus != nil`

Reply:
I don't think there is any harm doing both:
`if infra.Status.PlatformStatus != nil && infra.Status.PlaformStatus.Type == configapiv1.AWSPlatformType {`

Reply:
updated in next push